crates/bdk/tests/wallet.rs

```diff
@@ -1002,11 +1049,13 @@ fn test_add_foreign_utxo() {
         .only_witness_utxo()
         .add_foreign_utxo(utxo.outpoint, psbt_input, foreign_utxo_satisfaction)
         .unwrap();
-    let (mut psbt, details) = builder.finish().unwrap();
+    let mut psbt = builder.finish().unwrap();
+    let fee = psbt.fee_amount();
```

Comment:
:bulb: : We are using `psbt.fee_amount()` a lot in all of these tests but we also introduced our own fee calc method in this PR. Maybe we can test for free our one against the PSBT one with some macro like `let fee = check_fee!(wallet, psbt)` which would internally assert that psbt.fee_amount was the same as what the wallet calculates and then returns it.

Reply:
Good idea, I added it in https://github.com/bitcoindevkit/bdk/pull/1048/commits/f1d868d94ae02e6b30771ace01c84b3bed98c839, and was able to remove the test I added for checking fees with a foreign utxo that's now covered by the existing txbuilder foreign utxo tests.